tensorflow/core/kernels/sparse_matmul_op.cc

```diff
@@ -896,6 +897,8 @@ class LibxsmmSparseMatMul {
       } else {
         std::unique_ptr<TensorInfoCacheEntry> e{
             new TensorInfoCacheEntry{M, K, N, max_threads, {}, nullptr}};
+        // setup scoped allocator, which uses cpu_allocator() for this scope
+        const libxsmm_tf_allocator<libxsmm_scratch_allocator> tf_allocator;
```

Comment:
As far as I can tell, this ensures that all the calls that libxsmm makes to malloc use the tf_allocator at least for the duration of the tf_allocator variable.